Sources/Core/AWSClientRuntime/Sources/AWSClientRuntime/Protocols/AWSJSON/AWSJSONError.swift

```diff
@@ -11,7 +11,7 @@ import class SmithyHTTPAPI.HTTPResponse
 @_spi(SmithyReadWrite) import class SmithyJSON.Reader
 
 public struct AWSJSONError: BaseError {
-    public let code: String
+    public var code: String
```

Comment:
Why does this need to be set after the `AWSJSONError` is constructed?  Can't we pass `errorDetails` (or `nil` if not used) into the constructor?  Alternatively can we roll the `awsQueryCompatible` logic into this constructor & pass in a Bool to indicate whether that trait is present? (The purpose of the `BaseError` type is to encapsulate protocol-specific error logic like this.)

That would allow us to keep this property immutable.  Swift properties should generally be immutable unless there is a reason it must be changed after initialization.

Reply:
I didn't feel right about changing the constructor for the general AWSJsonError just for a service customization used only by SQS. Will think if there's a better way to keep this as a constant

Reply:
I'll have `code` passed in as an optional argument and then handle setting code in the extension function